### PR TITLE
Adapted to ES6 to avoid object mutation

### PR DESCRIPTION
**Pull request by GentooXativa**
Code adapted to ES6 to avoid object mutation, we are facing some troubles on newer systems, sometimes `config` from systemic is immutable and this leads to some errors on 3rd party librarys.

* Code adapted to ES6 to avoid object mutation
* Added `eslint` with `airbnb-base` ruleset
* Added fallback for `logger.info` to `console.log` when there is no logger dependency
* Dependency version updated to last stable version for `mongodb` and `lodash`

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# editorconfig.org
```
Comment (Betisman) [inline .editorconfig:1]: Do we need this file to be committed?

Comment (GentooXativa) [reply .editorconfig:1]: What's the difference between force eslint rules and use a editorconfig file to keep this settings on editors?

Both of them helps with the same issue, provide a consistent ruleset for code formatting

Comment (UlisesGascon) [reply .editorconfig:1]: +1 This file is useful to keep consistency in editors w/o complex plugins

**Review by UlisesGascon — APPROVED** [review]
LGTM! 

Awesome work @GentooXativa! 
This is a great upgrade for Systemic ecosystem 🎉

**Comment by UlisesGascon** [comment]
If you agree @Betisman I will merge the PR and bump the package version for NPM deploy as a Major 👍

**Comment by Betisman** [comment]
Let's roll @UlisesGascon

**Review by Betisman — APPROVED** [review]

File: .editorconfig (hunk 1 above)
Comment (Betisman) [reply .editorconfig:1]: If you change some rules in eslintrc but you don't do the same in editorconfig, what happens?

**Pull request merged**